modules/data.remote/R/GEDI_AGB_prep.R

```diff
@@ -0,0 +1,527 @@
+#' Prepare GEDI AGB data for the SDA workflow.
+#'
+#' @param site_info Bety list of site info including site_id, lon, and lat.
+#' @param time_points A vector contains each time point within the start and end date.
+#' @param outdir Where the final CSV file will be stored.
+#' @param buffer buffer distance for locate GEDI AGB searching box (default is 0.01 ~ 1 km)
+#' @param search_window search window for locate available GEDI AGB values.
+#'
+#' @return A data frame containing AGB and sd for each site and each time step.
+#' @export
+#' 
+#' @examples
+#' \dontrun{
+#' settings_dir <- "/projectnb/dietzelab/dongchen/anchorSites/SDA/pecan.xml"
+#' settings <- PEcAn.settings::read.settings(settings_dir)
+#' site_info <- settings %>% 
+#'   purrr::map(~.x[['run']] ) %>% 
+#'   purrr::map('site')%>% 
+#'   purrr::map(function(site.list){
+#'     #conversion from string to number
+#'     site.list$lat <- as.numeric(site.list$lat)
+#'     site.list$lon <- as.numeric(site.list$lon)
+#'     list(site_id=site.list$id, lat=site.list$lat, lon=site.list$lon, site_name=site.list$name)
+#'   })%>% 
+#'   dplyr::bind_rows() %>% 
+#'   as.list()
+#'   time_points <- seq(as.Date("2019-07-15"), as.Date("2023-07-15"), by = "1 year")
+#'   outdir <- "/projectnb/dietzelab/dongchen/anchorSites/GEDI_Tami/test/"
+#' GEDI_AGB <- GEDI_AGB_prep(site_info, time_points, outdir, buffer)
+#' }
+#' @author Dongchen Zhang
+#' @importFrom magrittr %>%
+GEDI_AGB_prep <- function(site_info, time_points, outdir = NULL, buffer = 0.01, search_window = "3 month") {
+  # if we don't have outdir, we will use the temp dir as outdir.
+  if (is.null(outdir)) {
+    outdir <- file.path(getwd(), "GEDI_AGB")
+    if (file.exists(outdir)) {
+      # if dir already exists.
+      # delete everything in this folder 
+      # to make sure the download process will work.
+      unlink(outdir, recursive = T)
```

Comment:
It was designed to avoid any pre-existing folder that happens to have the same name as `GEDI_AGB`. I think the user should identify this edge case by himself. Fixed.